src/screens/Departures/state/stop-place-details-state.ts

```diff
@@ -0,0 +1,135 @@
+import {useCallback, useEffect} from 'react';
+import useReducerWithSideEffects, {
+  NoUpdate,
+  ReducerWithSideEffects,
+  Update,
+  UpdateWithSideEffect,
+} from 'use-reducer-with-side-effects';
+import {ErrorType, getAxiosErrorType} from '@atb/api/utils';
+import {getStopsDetails} from '@atb/api/departures/stops-nearest';
+import {StopsDetailsQuery} from '@atb/api/types/generated/StopsDetailsQuery';
+
+export type StopsDetailsDataState = {
+  data: StopsDetailsQuery | null;
+  error?: {type: ErrorType};
+  ids?: Array<string>;
+  isLoading: boolean;
+};
+
+const initialState: StopsDetailsDataState = {
+  data: null,
+  error: undefined,
+  ids: undefined,
+  isLoading: false,
+};
+
+type StopsDetailsDataActions =
+  | {
+      type: 'LOAD_DETAILS';
+      locations?: Array<string>;
+    }
+  | {
+      type: 'STOP_LOADER';
+    }
+  | {
+      type: 'UPDATE_STOP_DETAILS';
+      ids?: Array<string>;
+      result: StopsDetailsQuery;
+    }
+  | {
+      type: 'SET_ERROR';
+      error: ErrorType;
+    };
+
+const reducer: ReducerWithSideEffects<
+  StopsDetailsDataState,
+  StopsDetailsDataActions
+> = (state, action) => {
+  switch (action.type) {
+    case 'LOAD_DETAILS': {
+      if (!action.locations) return NoUpdate();
+      return UpdateWithSideEffect<
+        StopsDetailsDataState,
+        StopsDetailsDataActions
+      >(
+        {
+          ...state,
+          isLoading: true,
+          error: undefined,
+        },
+        async (state, dispatch) => {
+          if (!action.locations) return;
+          try {
+            const result = await getStopsDetails({ids: action.locations});
+            dispatch({
+              type: 'UPDATE_STOP_DETAILS',
+              ids: action.locations,
+              result,
+            });
+          } catch (e) {
+            dispatch({
+              type: 'SET_ERROR',
+              error: getAxiosErrorType(e),
+            });
+          } finally {
+            dispatch({type: 'STOP_LOADER'});
+          }
+        },
+      );
+    }
+
+    case 'STOP_LOADER': {
+      return Update<StopsDetailsDataState>({
+        ...state,
+        isLoading: false,
+      });
+    }
+
+    case 'UPDATE_STOP_DETAILS': {
+      return Update<StopsDetailsDataState>({
+        ...state,
+        isLoading: false,
+        ids: action.ids,
+        data: action.result,
+      });
+    }
+
+    case 'SET_ERROR': {
+      return Update<StopsDetailsDataState>({
+        ...state,
+        error: {
+          type: action.error,
+        },
+        data: state.data,
+      });
+    }
+
+    default:
+      return NoUpdate();
+  }
+};
+
+/***
+ * Use state for fetching details for a list of stopPlaces
+ *
+ * @param {Array<string>} [locationIds] - NSR ids to fetch details for
+ */
+export function useStopsDetailsData(locationIds?: Array<string>) {
```

Comment:
Kunne `locationIds` vært entall `locationId` her?

Reply:
Tenker du bare navngiving, eller funksjonalitet her? Planlegger at samme state skal kunne brukes i situasjoner der det trengs mer enn én holdeplass, derav flertall.

Reply:
Mente endring i selve funksjonaliteten, basert på YAGNI. Tenk at hvis man hadde unit-testet hooken så måtte man nå testet både tom liste, liste med kun et element og liste med flere elementer. I praksis tester vi lokalt i simulator og under test-steget i prosessen, men da ender vi nok opp bare med å teste liste med et element, siden det er eneste implementasjon.

Reply:
Måtte slå opp den der 😅
> YAGNI: You aren't gonna need it

Spesifikt trenger vi en state for å laste detaljer for et sett med stoppested for favoritt tabben, https://github.com/AtB-AS/mittatb-app/issues/2427. Det er akkurat samme informasjon, bare at i dette tilfellet er det ett stopp, for favoritt tab for flere. Syns det gir mening å gjenbruke.

Reply:
Hvis du har en snarlig plan for det så er det greit :+1: